source/common/network/io_socket_handle_impl.cc

```diff
@@ -587,11 +587,19 @@ absl::optional<std::string> IoSocketHandleImpl::interfaceName() {
     return absl::nullopt;
   }
 
+  if (socket_address->ip()->version() != Address::IpVersion::v4 &&
+      socket_address->ip()->version() != Address::IpVersion::v6) {
+    ENVOY_BUG(false, fmt::format("unexpected IP family {}", socket_address->ip()->version()));
+  }
+
   Api::InterfaceAddressVector interface_addresses{};
   const Api::SysCallIntResult rc = os_syscalls_singleton.getifaddrs(interface_addresses);
   RELEASE_ASSERT(!rc.return_value_, fmt::format("getiffaddrs error: {}", rc.errno_));
 
   absl::optional<std::string> selected_interface_name{};
+  absl::uint128 socket_address_value = socket_address->ip()->version() == Address::IpVersion::v4
```

Comment:
I think the prior version would return an empty interface name but I think falling through here might segfault instead.  Maybe add a unit test?